test/langtools/tools/javap/TestClassNameWarning.java

```diff
@@ -172,14 +176,15 @@ public void testNoNameClass(Path base) throws Exception {
                 .files(tb.findJavaFiles(src))
                 .run()
                 .writeAll();
-
-        ClassFile cf = ClassFile.read(classes.resolve("A.class"));
-        ClassFile cf2 = new ClassFile(
-                cf.magic, cf.minor_version, cf.major_version, cf.constant_pool,
-                cf.access_flags,
-                0, // this_class,
-                cf.super_class, cf.interfaces, cf.fields, cf.methods, cf.attributes);
-        new ClassWriter().write(cf2, Files.newOutputStream(classes.resolve("Z.class")));
+        ClassModel cm = Classfile.of().parse(classes.resolve("A.class"));
+        Classfile.of().buildTo(
+                classes.resolve("Z.class"),
+                ClassDesc.of("0"), cb -> {
```

Comment:
Zero constant pool index means there is physically no entry (a kind of null value).
According to the spec the class name CP entry is mandatory, so Classfile API does not allow to enter no value and create a class file with 0 index of the class name.
We will have to figure out the way how to write 0 CP index for class name or skip this part of the test.
ClassDesc.of("0") is valid class name, so testing of the 0 CP index is skipped here.

Reply:
Classfile API doesn't allow to create an invalid entry.
I'll investigate if it is worth to extend the API, implement some low-lever workaround or skip this specific test case and leave it with valid class name.

Reply:
I would suggest to remove the whole testNoNameClass test case as the warning message is tested by  similar testWrongNameClass test case below. Testing specifically for 0 class name CP index is obsolete.